pkg/controller/che/che_controller.go

```diff
@@ -150,6 +151,49 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	var toRequestMapper handler.ToRequestsFunc = func(obj handler.MapObject) []reconcile.Request {
+		checlusters := &orgv1.CheClusterList{}
+		if err := mgr.GetClient().List(context.TODO(), &client.ListOptions{}, checlusters); err != nil {
+			return []reconcile.Request{}
+		}
+
+		if len(checlusters.Items) != 1 {
+			return []reconcile.Request{}
+		}
+
+		if checlusters.Items[0].Spec.Server.ServerTrustStoreConfigMapName != obj.Meta.GetName() {
+			return []reconcile.Request{}
+		}
+
+		return []reconcile.Request{
+			reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: checlusters.Items[0].Namespace,
+					Name:      checlusters.Items[0].Name,
+				},
+			},
+		}
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: toRequestMapper,
+	}, predicate.Funcs{
```

Comment:
I assume that this predicate could be extracted as something like the `OnAllExceptGenericEvents` predicate.

Reply:
Done